### PR TITLE
Redirect to Login and Back

### LINKED ISSUE #632: Check routes for "if logged in"
**Is your feature request related to a problem? Please describe.**
Some route may be viewable or show errors if the user is not logged in when they should be.

**Describe the solution you'd like**
Bounce the user back to the login page to make them login first. Perhaps, remember what link they were trying to access before hand, and once logged in, send them directly to that link.

**Additional context**

### PR DESCRIPTION
**Pull request by JMozes16**
### Description of change(s), including why:

Previously, anyone could go to any link without being logged in. I made it so that if you are not logged in, you get redirected to the login page, and then once logged in, it redirects you back to the previous page that you tried to go to prior to logging in.

### Type of change(s)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Any work to still do or other concerns:

- (none)

### Closing issues:

- Closes #632

### Screenshots (if frontend change) of before and after:
Using /groups route as an example, this is functionally the same for all other routes that should require login:
When not logged in and going to /groups route before:
groupsPage

When not logged in and going to /groups after(redirects to /login):
loginPage

Then if logged in successfully, redirects back to /groups:
groupsPage

### CODE REVIEW

File: PollBuddy-Server/frontend/src/components/Dropdown/Dropdown.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ function useOutsideAlerter(ref, menuProps) {
           console.log(response);
           if(response.result === "success") {
             //Logout has succeeded, Clear frontend user data
-            localStorage.setItem("loggedIn", false);
+            localStorage.setItem("loggedIn", "false");
```
Comment (hbh7) [inline PollBuddy-Server/frontend/src/components/Dropdown/Dropdown.js:54]: I'm also curious as to why you changed this to a string.

File: PollBuddy-Server/frontend/src/pages/AccountInfo/AccountInfo.js [hunk 2]
```diff
@@ -251,6 +244,9 @@ class AccountInfo extends Component {
   }
 
   render() {
+    if (localStorage.getItem("loggedIn") !== "true") {
```
Comment (hbh7) [inline PollBuddy-Server/frontend/src/pages/AccountInfo/AccountInfo.js:247]: Wouldn't this be redundant with the `PrivateComponent` added in through `app.js`?

**Review by hbh7 — COMMENTED** [review]
Interesting implementation, I'm curious as to why you pass in `state = {true}` to PrivateComponent and never anything else. What function does doing that serve? My guess is something related to line 27 in `LoginDefault.js`, for example. I've left a few other comments, mostly just trying to make sure I understand how this works for when people ask me later lol. If you wouldn't mind leaving some comments in the code about what's happening too, that'd be much appreciated!

**Comment by JMozes16** [comment]
> 
Passing in state = {true} to PrivateCompontent is saying that you must be logged in to access this route. Inside of PrivateComponent, it will check what to do based on if the user is actually logged in or not.  I have now updated app.js so that the login and register routes have state = {false} passed into PrivateComponent. So logged in users will be redirected to the homepage when trying to access these routes. They must logout before going to them.

File: PollBuddy-Server/frontend/src/components/Dropdown/Dropdown.js (hunk 1 above)
Comment (JMozes16) [reply PollBuddy-Server/frontend/src/components/Dropdown/Dropdown.js:54]: This is how it is actually stored, it was not working before I changed it to a string.

File: PollBuddy-Server/frontend/src/pages/AccountInfo/AccountInfo.js (hunk 2 above)
Comment (JMozes16) [reply PollBuddy-Server/frontend/src/pages/AccountInfo/AccountInfo.js:247]: Yup, just removed it.

Comment (JMozes16) [reply PollBuddy-Server/frontend/src/pages/AccountInfo/AccountInfo.js]: I think I was just using it for debugging and forgot to remove it after, thanks!

**Review by hbh7 — APPROVED** [review]
Sounds good, thanks for the clarification

**Pull request merged**